app/src/androidTest/java/org/mozilla/fenix/helpers/MatcherHelper.kt

```diff
@@ -0,0 +1,85 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.helpers
+
+import androidx.test.uiautomator.UiObject
+import androidx.test.uiautomator.UiSelector
+import org.junit.Assert.assertTrue
+import org.mozilla.fenix.helpers.TestAssetHelper.waitingTime
+import org.mozilla.fenix.helpers.TestHelper.mDevice
+
+/**
+ * Helper for querying and interacting with items based on their matchers.
+ */
+object MatcherHelper {
+
+    fun appItemWithResourceId(resourceId: String) =
+        mDevice.findObject(UiSelector().resourceId(resourceId))
+
+    fun appItemContainingText(itemText: String) =
+        mDevice.findObject(UiSelector().textContains(itemText))
+
+    fun appItemWithDescription(description: String) =
+        mDevice.findObject(UiSelector().descriptionContains(description))
+
+    fun appItemStateWithResourceId(resourceId: String, state: Boolean) =
+        mDevice.findObject(UiSelector().resourceId(resourceId).checked(state))
+
+    fun appItemStateWithResourceIdAndText(resourceId: String, text: String, state: Boolean) =
```

Comment:
I think replacing the "state" parameter with "isChecked: boolean" will add more clarity when using these methods (without looking at their definition). Same for the "state" part in the method name. Could be confused with the "exists" state check below.

Also, a small change, optional: I think you can remove the "app" part from all these matchers names

Reply:
Same, maybe add an if else statement if the element is not checked?

Reply:
> Same, maybe add an if else statement if the element is not checked?

If it's not checked you just pass `false` for `isChecked` from `checkedItemWithResIdAndText()`